### DISCUSSION TITLE
Utilize prettier's `printWidth` behaviour

### DISCUSSION THREAD

**Discussion by codeflorist** [antfu/eslint-config#486] [Q&A]
Is it possible with this package to utilize prettier's `printWidth` functionality to automatically wrap e.g. HTML tags in vue-files, if the lines become too long?

I know there is the `vue/max-attributes-per-line` rule, it doesn't take the line's length in account - thus unnecessarily wrapping perfectly readable tags with short attributes - and possibly not wrapping tags with few but very long attributes.

**Comment by TotomInc** [comment]
I'm exactly in the same boat as you, except it is for the JSX part when used with React.

I would like to use the `style/jsx-max-props-per-line` rule with a `style/max-len` combined with Prettier's `printWidth` functionnality.

I actually haven't found a successful solution, and I believe this discussion should instead live on the ESLint Stylistic repository.

**Reply by codeflorist** [reply]
This package utilizes `eslint-plugin-format`, which in turn uses `prettier` itself. So `prettier` is already at work here. That's why i thought, this might already be possible or could be easily added as a feature.

Of course, if something similar is added to ESLint Stylistic, then that's also good.

**Reply by TotomInc** [reply]
Nice catch. I spent some time integrating `eslint-plugin-prettier` and manually detecting Stylistic rules that conflicts with Antfu's config.

A large part of this work can be seen here if that can help you: https://github.com/TotomInc/eslint-config-next/commit/258f2080a88b1ac28ca2b38b7c3ba0306101def2

However I don't think it's worth to use `eslint-plugin-format` instead of the official `eslint-plugin-prettier`, which seems to add another layer of abstraction into the config.

I would like to add that I ended up with integrating Prettier because I am used to its formatting which cannot be reproduced by any actual ESLint plugin.

**Comment by Psycarlo** [comment]
+1

Btw, antfu's eslint-config is aimed to be used standalone without Prettier
